server/channels/app/post_metadata.go

```diff
@@ -362,6 +362,11 @@ func (a *App) getImagesForPost(c request.CTX, post *model.Post, imageURLs []stri
 	}
 
 	for _, imageURL := range imageURLs {
+		// MM-51060 - don't accept Mattermost permalinks as Open Graph Image URLs.
```

Comment:
Instead of disabling _all permalinks_, can we verify if it's solved by disabling just one permalink like `if permalink == post.permalink then continue;`?